Formula/glui.rb

```diff
@@ -30,6 +31,7 @@ class Glui < Formula
   end
 
   def install
+    ENV.append "CXXFLAGS", "-fPIE" if OS.linux?
```

Comment:
I'm a bit suspicious of this. If this is needed to make the library usable, upstream should set this by default. What do other distros with the same/newer versions of `ld` do? It's probably worth filing an issue upstream about this.

Reply:
Arch Linux and Fedora take a different approach and patch the Makefile to support building a shared library: https://aur.archlinux.org/cgit/aur.git/tree/makefile.patch?h=glui, https://src.fedoraproject.org/rpms/glui/blob/rawhide/f/glui-2.36-solib.patch.  These patches would need some modification to also support macOS, but we could then submit the patch to the upstream GitHub.   

Alternatively I can just make a PR to add `-fPIE` to the CPPFLAGS in the Makefile, though I'm not sure if that would cause issues on some platforms if it was always on.  Or the simplest case is to just report the issue.  

One complication in all of this is the the version of `ld` isn't the whole story - we would previously have been using `ld` from our `binutils`, which is now at version `2.39`, while Ubuntu 22.04 uses `2.38`.  So the versions are comparable if not newer.  The problem is that Linux distros configure their linkers in different ways.  For example, Debian/Ubuntu uses `--as-needed`, while Fedora/Red Hat (and also Arch Linux I think) don't use `--as-needed`.  It's terribly confusing at times because it can mean that the exact same linking step on one distro will work with no issues but will fail with arcane and cryptic errors on another distro.

Interestingly that may imply that another solution to some of these linking issues is to add brewed `binutils` as a build-time dependency so we can use our linker instead of the host one.  I haven't looked into this but it might provide additional flexibility in some of these scenarios.

Reply:
Can we strip `--as-needed` flags in superenv? Or, at least, find a way to avoid passing it unless requested by the project?

Reply:
That's something I haven't had a chance to look into yet, but would be a good idea.  It's not entirely clear to me yet if `--as-needed` is the culprit but it's quite possible.  I have been able to drop `--as-needed` using `CMAKE` using https://cmake.org/cmake/help/latest/prop_tgt/LINK_WHAT_YOU_USE.html, but I haven't explored the approach for other build systems.

Reply:
After doing more digging, it turns out there that this probably should just be solved by a revision bump.  As of Ubuntu 17.04, `-fPIE`/`-fPIC` is mandatory for all binaries.  The existing bottle will fail its test but if I am right, simply rebuilding it in Ubuntu 22.04 will fix the issue with no changes to the formula.